src/stores/ScannerStore.js

```diff
@@ -390,16 +450,22 @@ export default class ScannerStore extends Container<ScannerState> {
 	}
 
 	cleanup() {
-		this.setState(defaultState);
+		return new Promise(resolve => {
```

Comment:
Why this is wrapped into `Promise`? there are all synchronous functions I think.

Reply:
this is a workaround to an edge case Thibaut discovered in the second part of this: https://github.com/paritytech/parity-signer/pull/428#pullrequestreview-305117130

It is to make sure the state is set before the next frame is scanned. In this case the setState is asynchronous.

Reply:
actually in all cases `setState` is asynchronous, it just usually doesn't matter.

Reply:
Ah, yeah, set state return a promise here, you can do `await this.setStatte`, which is the same I think.